schema.graphql

```diff
@@ -1,69 +1,98 @@
-  collectionItemKey: String! @index
-  collection: Collection!
-  metadataCid: String
-  isBurned: Boolean!
-  createdAt: Float @index
-  updatedAt: Float @index
-  owner: String
+type Item @entity @compositeIndexes(fields: [["owner", "itemId"]]) {
+    id: ID!
+    itemId: Int! @index
+    collectionItemKey: String! @index
+    collection: Collection!
+    metadataCid: String
+    isBurned: Boolean!
+    createdAt: Int @index
+    updatedAt: Int @index
+    owner: String
 }
 
-type BalanceTransfer @entity {
-  id: ID!
-  from: String! @index
-  to: String! @index
-  txHash: String @index
-  amount: BigInt @index
-  timestamp: Float @index
-  blockNumber: BigInt
-  success: Boolean
+type BalanceTransfer @entity @compositeIndexes(fields: [["from", "to"]]) {
+    id: ID!
+    from: String! @index
+    to: String! @index
+    txHash: String @index
+    amount: BigInt @index
+    timestamp: Int @index
+    blockNumber: BigInt
 }
 
-type AllocationEventData @jsonField {
-  who: String @index
-  amount: String
+type Rewards @entity @compositeIndexes(fields: [["from", "to"]]) {
+    id: ID!
+    from: String! @index
+    to: String! @index
+    txHash: String @index
+    amount: BigInt @index
+    timestamp: Int @index
+    blockNumber: BigInt
 }
 
-type AllocationEvent @jsonField {
-  method: String @index
-  section: String @index
-  index: String
-  data: AllocationEventData
+type TransferToTreasury @entity @compositeIndexes(fields: [["from", "to"]]) {
+    id: ID!
+    from: String! @index
+    to: String! @index
+    txHash: String @index
+    amount: BigInt @index
+    timestamp: Int @index
+    blockNumber: BigInt
+    isAllocation: Boolean @index
 }
 
-type Allocations @entity {
-  id: ID!
-  txHash: String @index
-  block: Int @index
-  data: [AllocationEvent]
-  success: Boolean
+type UniquesTransfer @entity @compositeIndexes(fields: [["from", "to"]]) {
+    id: ID!
+    txHash: String @index
+    block: Int @index
+    from: String @index
+    to: String @index
+    collection: Collection!
+    item: Item!
+    fee: BigInt
+    timestamp: Int @index
+    success: Boolean
 }
 
-type UniquesTransfer @entity {
-  id: ID!
-  txHash: String @index
-  block: Int @index
-  from: String @index
-  to: String @index
-  collection: Collection!
-  item: Item!
-  fee: BigInt
-  timestamp: Float @index
-  success: Boolean
+type Pot @entity {
+    id: ID!
+    potId: Int! @index
+    owner: String! @index
+    feeQuota: BigInt!
+    reserveQuota: BigInt!
+    sponsorshipType: String!
+    sponsoredAccounts: [AccountPotBalance] @derivedFrom(field: "pot")
+    createdAt: Int @index
+    updatedAt: Int @index
+}
+
+type AccountPotBalance @entity @compositeIndexes(fields: [["pot", "account"]]) {
+    id: ID!
+    pot: Pot!
+    account: Account!
+    feeQuota: BigInt!
+    reserveQuota: BigInt!
+    createdAt: Int @index
+    updatedAt: Int @index
+}
+
+type Account @entity {
+    id: ID!
+    potBalance: [AccountPotBalance] @derivedFrom(field: "account")
```

Comment:
I'm a bit worried about the circular kind of dependency from AcountPotBalance to Account and then from Account to AccountPotBalance